identity-service/test/configTest.js

```diff
@@ -0,0 +1,150 @@
+const assert = require('assert')
+const config = require('../src/config')
+const { generateRandomUUID, generateRandomPort, generateRandomNaturalNumber, generateRandomBoolean, generateRandomLogLevel } = require('./lib/generateRandomValues')
+const convict = require('convict')
+
+describe('test txRelay functions', function () {
+  it('env key and object key names are equal', function () {
+    const schema = config.getSchema().properties
+
+    for (var key in schema) {
+      assert(key === schema[key]['env'], `The key ${key} and the env key ${schema[key]['env']} do not match.`)
+    }
+  })
+
+  it('[pass] Adding a number with format as type string will cause config to take its string value', function () {
+    const c = convict({
+      randomKey: {
+        format: String,
+        default: null,
+        env: 'randomKey'
+      }
+    })
+
+    process.env['randomKey'] = 123
+    assert(process.env['randomKey'] === '123')
+
+    c.load({})
+
+    assert(c.get('randomKey') === '123')
+
+    try {
+      c.validate()
+    } catch (e) {
+      assert.fail(e)
```

Comment:
if we hit an error here, the test is designed to fail?

Reply:
Yes, added a comment for clarity